wsdlgo/encoder.go

```diff
@@ -601,35 +632,72 @@ var isGoKeyword = map[string]bool{
 }
 
 type parameter struct {
-	code     string
-	xmlToken string
+	Name     string
+	Type     string
+	XMLName  string
+	XMLToken string
+}
+
+func (p parameter) asGo() string {
+	c := p.Name + " " + p.Type
+	return c
+}
+
+func (p parameter) goDefault() string {
+	v := trimns(p.Type)
+	if v != "" && v[0] == '*' {
+		v = v[1:]
+	}
+	switch v {
+	case "error":
+		return `errors.New("not implemented")`
+	case "bool":
+		return "false"
+	case "int", "int64", "float64":
+		return "0"
+	case "big.Float":
+		return "big.Float{}"
+	case "string":
+		return `""`
+	case "[]byte":
+		return "nil"
+	default:
+		return "&" + v + "{}"
+	}
 }
 
-func code(list []*parameter) []string {
-	code := make([]string, len(list))
-	for i, p := range list {
-		code[i] = p.code
+func asGoParamsString(params []*parameter) string {
+	goP := make([]string, len(params))
+	for i, p := range params {
+		goP[i] = p.asGo()
 	}
-	return code
+	return strings.Join(goP, ", ")
 }
 
-func (ge *goEncoder) genParams(m *wsdl.Message, needsTag bool) []*parameter {
-	params := make([]*parameter, len(m.Parts))
-	for i, param := range m.Parts {
+func (ge *goEncoder) genParams(parts []*wsdl.Part, needsTag bool) []*parameter {
+	params := make([]*parameter, len(parts))
+	for i, part := range parts {
 		var t, token string
 		switch {
-		case param.Type != "":
-			t = ge.wsdl2goType(param.Type)
+		case part.Type != "":
+			t = ge.wsdl2goType(part.Type)
 			token = t
-		case param.Element != "":
-			t = ge.wsdl2goType(param.Element)
-			token = ge.trimns(param.Element)
+		case part.Element != "":
+			t = ge.wsdl2goType(part.Element)
+			token = trimns(part.Element)
 		}
-		name := param.Name
+		// Go fields and variable names generate fewer warnings with "ID" instead of "Id"
+		name := regexp.MustCompile("(.*)Id$").ReplaceAllString(part.Name, "${1}ID")
```

Comment:
can we not compile this regex on every param generation? i'd declare it global right above the genParams func